reconcile/openshift_limitranges.py

```diff
@@ -0,0 +1,109 @@
+import logging
+import semver
+import sys
+
+import utils.gql as gql
+import reconcile.openshift_base as ob
+
+from utils.openshift_resource import OpenshiftResource as OR
+from utils.defer import defer
+
+from reconcile.queries import NAMESPACES_QUERY
+
+QONTRACT_INTEGRATION = 'openshift-limitranges'
+QONTRACT_INTEGRATION_VERSION = semver.format_version(0, 1, 0)
+
+SUPPORTED_LIMITRANGE_TYPES = (
+    'default',
+    'defaultRequest',
+    'max',
+    'maxLimitRequestRatio',
+    'min'
+)
+
+
+def construct_resources(namespaces):
+    for namespace in namespaces:
+        # Get the linked limitRanges schema settings
+        limitranges = namespace.get("limitRanges", {})
+
+        body = {
+            'apiVersion': 'v1',
+            'kind': 'LimitRange',
+            'metadata': {
+                'name': limitranges['name'],
+            },
+            'spec': {
+                'limits': []
+            }
+        }
+
+        # Build each limit item ignoring null ones
+        speclimit = {}
+        for l in limitranges['limits']:
+            for ltype in SUPPORTED_LIMITRANGE_TYPES:
+                if ltype in l and l[ltype] is not None:
+                    speclimit[ltype] = l[ltype]
+            body['spec']['limits'].append(speclimit)
+
+        resource = OR(body, QONTRACT_INTEGRATION, QONTRACT_INTEGRATION_VERSION)
+
+        # Create the resources and append them to the namespace
+        namespace["resources"] = [resource]
```

Comment:
i really like it that you modify the existing namespace instead of creating a new object. that makes a lot of sense and i will start doing it as well!
its also good that you are creating a list, following this comment:
https://gitlab.cee.redhat.com/service/app-interface/merge_requests/2039#note_967595

does that make sense?

Reply:
We want to take over the limit ranges on namespaces we control. I thought about allowing more than one to be created (ie. a list instead of a single reference) but I could not think of a use case.

I'll leave it as is for now and we can revisit if needed in the future

Reply:
sgtm